### DISCUSSION TITLE
Feature Request: New auto menu placement option

### DISCUSSION THREAD

**Discussion by dav-sap** [JedWatson/react-select#4442] [Ideas]
**Feature Request**

*   What's your use case?
    When theres no room on bottom I don't want the `Menu` to just move to top, I would like it to         just move a bit to the top, it can hide the `Control`, it will just fit in remaining space till bottom

*   What interface do you have in mind?
     Add option to `menuPlacement` prop

*   Can you point to similar functionality with any existing libraries or components? 
     Material UI Select does this, it hides the control if needs and just fills the room left in the    bottom.

**Comment by ebonow** [comment]
Sorry for the late reply to this, but wanted to better understand the ask. To better illustrate the request, it appears that Material-UI places the menu directly over the control.

Here is an unfocused select
<img width="353" alt="Screen Shot 2021-02-10 at 10 03 19 AM" src="https://user-images.githubusercontent.com/1007579/107551651-625dbf00-6b87-11eb-9cfa-bf81808e3296.png">

Here is the default open menu positioning 
<img width="231" alt="Screen Shot 2021-02-10 at 10 03 25 AM" src="https://user-images.githubusercontent.com/1007579/107551663-62f65580-6b87-11eb-92ee-9e37a3aca727.png">

Here is the menu positioning when opened near the bottom of the menu
<img width="308" alt="Screen Shot 2021-02-10 at 10 03 38 AM" src="https://user-images.githubusercontent.com/1007579/107551667-638eec00-6b87-11eb-8524-ee2df088a3ee.png">

[Source](https://material-ui.com/components/selects/#customized-selects) 

Converted from an issue to a discussion feature request...
